.github/workflows/run-tests.yml

```diff
@@ -13,6 +13,54 @@ on:
       commit:
         description: "Commit SHA"
         required: true
+      run_single_functional_test:
+        description: "Run a single functional test"
+        type: boolean
+        default: false
+      run_single_unit_test:
+        description: "Run a single unit test (INSTEAD of functional test)"
+        type: boolean
+        default: false
+      unit_test_directory:
+        description: "[Unit Test Only] Directory to run unit tests in"
+        type: string
+        default: "./temporal"
+      n_runs:
+        description: "Number of times to repeat the single test per database type"
+        type: number
+        default: 1
+      test_name:
+        description: "Name of the test to run (i.e. 'TestAcquireShard_DeadlineExceededErrorSuite' or 'TestFunctionalSuite/TestUpdateWorkflow')"
+        type: string
+      timeout_minutes:
+        description: "Test timeout in minutes"
+        type: number
+        default: 120
+      test_runner:
+        description: "Which runner to use. Choose higher RAM if your n_runs is high."
+        type: choice
+        default: "16GB RAM (ubuntu-20.04)"
+        options:
+          - "16GB RAM (ubuntu-20.04)"
+          - "64GB RAM (ubuntu-22.04)"
+      cassandra:
+        description: "cassandra"
+        type: boolean
+        default: false
+      sqlite:
+        description: "sqlite"
+        type: boolean
+        default: false
+      #      mysql8:
+      #        description: "mysql8"
+      #        type: boolean
+      #        default: false
+#      postgres12:
+#        description: "postgres12"
+#        type: boolean
+#        default: false
+          # if [[ "${{ inputs.mysql8 }}" == "true" ]]; then dbs_arr+=("mysql8"); fi
+        # if [[ "${{ inputs.postgres12 }}" == "true" ]]; then dbs_arr+=("postgres12"); fi
```

Comment:
The mysql8 and postgres db options are commented out because github limits action inputs to 10 max.
I omitted these two db options to add the unit test options.
Any ideas for omitting inputs to get the total down to 10 are welcome! I think the `test_runner` input could be omitted, since I didn't really notice a benefit to the 64GB machine (I got oomkills on both).

The `timeout_minutes` could also maybe be omitted.

Or, if the correct answer is to take the dbs as one string list instead of 4 booleans, I could refactor to do that too, I just like the "UI" of getting to select which dbs you want to run on via checkboxes.

Reply:
https://medium.com/@hemantbhosale916/github-actions-alternative-to-define-more-than-10-inputs-for-workflow-dispatch-event-d4d327d2680b here they get around the limit by taking in a JSON input

Reply:
some more discussion of the input limit here: https://github.com/actions/runner/issues/2076

Reply:
I made it a list instead of booleans